### ISSUE TITLE
[bitnami/common] `common.tplvalues.merge` ignores precedence on falsy values

### ISSUE THREAD

**Issue by maxnitze** [bitnami/charts#29566]
### Name and Version

bitnami/common 2.23.0

### What architecture are you using?

None

### What steps will reproduce the bug?

1. Have two dicts with the following values in your chart:

```yaml
dict1:
  enabled: true
  name: 1001
  myList:
    - "value1"
  someOtherKey: "someOtherValue"

dict2:
  enabled: false
  name: 1234
  myList: []
  anotherKey: "anotherValue"
```

2. Call the `common.tplvalues.merge` on the two dicts

```gotpl
{{- include "common.tplvalues.merge" (dict "values" (list .Values.dict2 .Values.dict1) "context" $) }}
```

EDIT: swapped the two dicts in this example, since the precedence in the helper is "from left to right". So "earlier" dicts have higher precedence than "later" ones (which is quite un-intuitive to me; see https://github.com/bitnami/charts/issues/29566#issuecomment-2382986146).

### What is the expected behavior?

I would expect to have a merged context like this:

```yaml
mergedContext:
  enabled: false
  name: 1234
  myList: []
  someOtherKey: "someOtherValue"
  anotherKey: "anotherValue"
```

### What do you see instead?

The following merged context:

```yaml
mergedContext:
  enabled: true
  name: 1234
  myList:
    - "value1"
  someOtherKey: "someOtherValue"
  anotherKey: "anotherValue"
```

### Additional information

Stumbled upon this issue while working on #27628 .

I'm not entirely sure if this is a bug or "dy-design". The docs state, that precedence in the helper is "consistent with http://masterminds.github.io/sprig/dicts.html#merge-mustmerge".

If this is by-design (or should not be fixed due to backwards compatibility reasons), I would propose to add a new helper, that has the behavior I would expect from this one already:

```gotpl
{{/*
Merge a list of values that contains template after rendering them.
Later values taking higher precedence. So merging two dicts with values `true` and `false` for the same key will result
in the value of the latter on that key. When using the default merge functionality this will always result in `true`.
Usage:
{{ include "common.tplvalues.merge-with-precedence" ( dict "values" (list .Values.path.to.the.Value1 .Values.path.to.the.Value2) "context" $ ) }}
*/}}
{{- define "common.tplvalues.merge-with-precedence" -}}
{{- $dst := dict -}}
{{- range .values -}}
{{- $val := include "common.tplvalues.render" (dict "value" . "context" $.context "scope" $.scope) | fromYaml }}
{{- range $key := keys $val }}
{{- $_ := set $dst $key (get $val $key) }}
{{- end }}
{{- end -}}
{{ $dst | toYaml }}
{{- end -}}
```

**Comment by andresbono** [comment]
Hi @maxnitze, thank you very much for taking the time to open this issue. Adding a new helper, as you are proposing, sounds good 👍. Please, feel free to send a PR with your suggestion.

Thank you again!

**Comment by maxnitze** [comment]
While creating the PR I checked the docs of the original helper `common.tplvalues.merge` again and I think I got some things mixed up.

The examples are not working as shown above (I think I just pasted the wrong ones). Actually the **first** dict gets precedence when using the `sprig/merge` function:

> Merge two or more dictionaries into one, giving precedence to the dest dictionary:
> 
> `$newdict := merge $dest $source1 $source2`

see https://masterminds.github.io/sprig/dicts.html#merge-mustmerge

I'll update the original post (by just swapping the two original dicts).

The original issue still stays the same though. I also found an issue in the `Masterminds/sprig` GitHub for this: https://github.com/Masterminds/sprig/issues/134

As stated there, the fix for this is to use the `sprig/mergeOverwrite` function:

> Merge two or more dictionaries into one, giving precedence from right to left, effectively overwriting values in the dest dictionary:
> 
> Given: 
> 
>     dst:
>       default: default
>       overwrite: me
>       key: true
>     
>     src:
>       overwrite: overwritten
>       key: false
> 
> will result in:
> 
> 
>     newdict:
>       default: default
>       overwrite: overwritten
>       key: false
> 
> `$newdict := mergeOverwrite $dest $source1 $source2`

see https://masterminds.github.io/sprig/dicts.html#mergeoverwrite-mustmergeoverwrite

So instead of my custom code I'll opt for using this already existing sprig function in the PR.
